Ennemy development

Hendrik-Kruger opened the pull request:
creation of enemies

Hendrik-Kruger committed:
Imported enemy prototypes

Hendrik-Kruger committed:
Made enemy movement and collision areas for the  ribbug

matthewwarn reviewed (changes requested):
Collision zone for detection is probably a good range left to right, but it should extend more vertically. Because right now, when the player jumps, they are out of the collision and the monster stops moving entirely. 

Also the walking animation needs to be fixed. Looks like its including a row of pixels from above it in the sprite sheet.

And maybe just add a killzone to the enemy hitbox for now. More specific attacking and whatnot could be added later, but for now it would be good if they just had an insta-kill zone.

Hendrik-Kruger committed:
added global node worked on enemy added hitboxes and animations to player and enemy

global will for now track if player is attacking
hitboxes are just to detect if enemy or player are able to attack and take damage

Hendrik-Kruger committed:
undid some changes to resolve confflics

matthewwarn committed:
Merge branch 'main' into Ennemy_development
